app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Section.scala

```diff
@@ -46,11 +48,34 @@ object Section {
     repeatsMax: Option[Expr],
     pages: NonEmptyList[Page],
     addAnotherQuestion: FormComponent,
-    instruction: Option[Instruction]
+    instruction: Option[Instruction],
+    presentationHint: Option[List[PresentationHint]]
   ) extends Section {
     override lazy val expandedFormComponents: List[FormComponent] = pages.toList.flatMap(_.expandedFormComponents)
   }
 
+  object AddToList {
+
+    val reads: Reads[AddToList] = (
+      (JsPath \ "title").read[SmartString] and
+        (JsPath \ "description").read[SmartString] and
+        (JsPath \ "shortName").read[SmartString] and
+        (JsPath \ "includeIf").readNullable[IncludeIf] and
+        (JsPath \ "repeatsMax").readNullable[Expr] and
+        (JsPath \ "pages").read[NonEmptyList[Page]] and
+        (JsPath \ "addAnotherQuestion").read[FormComponent] and
+        (JsPath \ "instruction").readNullable[Instruction] and
+        (JsPath \ "presentationHint").read[Option[List[PresentationHint]]](
+          (json: JsValue) =>
+            parseOpt(JsDefined(json), PresentationHintParser.validate)
+              .fold(us => JsError(us.toString), fv => JsSuccess(fv)))
+    )(AddToList.apply _)
```

Comment:
I don't like this explicit `Reads`. Cannot we just define `Reads[PresentationHint]` which will do the parsing?

Reply:
Thanks @josef-vlach. I have replaced this with a `Reads[List[PresentationHint]]`. Have made it a list to mimic what is done on FormComponent level